### PR TITLE
HS-1012 Fixed the IpInterface and SnmpInterface schema and data model

### PR DESCRIPTION
**Pull request by yangli2014**
## Description

Updated the schema and data model, the node scan result data model, the scan process, and the results process.
## Jira link(s)
- https://issues.opennms.org/browse/HS-1012

## Checklist
* [x] Follows Horizon Stream's [development guidelines.](https://github.com/OpenNMS/horizon-stream/wiki/Development-Guidelines)
* [x] Appropriate reviewer(s) have been selected.
* [x] Jira issue(s) have been updated to "In Review".
* [x] Includes [appropriate tests.](https://github.com/OpenNMS/horizon-stream/wiki/Test-Strategy)
* [ ] Documentation has been updated as necessary.

### CODE REVIEW

File: inventory/src/test/java/org/opennms/horizon/inventory/repository/IpInterfaceRepositoryTest.java [hunk 1]
```diff
@@ -64,12 +66,21 @@ class IpInterfaceRepositoryTest {
...
+    private SnmpInterfaceRepository snmpInterfaceRepository;
 
     @BeforeEach
     public void setup() throws UnknownHostException {
         loadNodes();
     }
 
+    @AfterEach
+    public void cleanUp() {
+        nodeRepository.deleteAll();
```
Comment (james8hut) [inline inventory/src/test/java/org/opennms/horizon/inventory/repository/IpInterfaceRepositoryTest.java:79]: Tenancy has changed due to HS-1013.
Without code like the below, nothing will be deleted:
        Context.current().withValue(GrpcConstants.TENANT_ID_CONTEXT_KEY, tenantId).run(()->
        {
            nodeRepository.deleteAll();
        });

File: inventory/src/test/java/org/opennms/horizon/inventory/repository/IpInterfaceRepositoryTest.java [hunk 2]
```diff
@@ -98,6 +109,53 @@ void testFindByIpInterfaceForAGivenLocationAndIpAddress() throws UnknownHostExce
...
+    @Test
+    void testIpInterfaceWithSNMPIfDeleteSNMP() throws UnknownHostException {
+        int ifIndex = 10;
+        var node = nodeRepository.findByNodeLabel("node1").get(0);
+        var snmpInterface = new SnmpInterface();
+        snmpInterface.setNode(node);
+        snmpInterface.setIfIndex(ifIndex);
+        snmpInterface.setIfName("test-snmp");
+        snmpInterface.setTenantId(node.getTenantId());
+        snmpInterfaceRepository.save(snmpInterface);
```
Comment (james8hut) [inline inventory/src/test/java/org/opennms/horizon/inventory/repository/IpInterfaceRepositoryTest.java:121]: Similarily here:
In this case, you'll get an error about the tenantId being different to the one it is trying to save.
Best to wrap the entire method with the code above

File: inventory/src/test/java/org/opennms/horizon/inventory/repository/IpInterfaceRepositoryTest.java [hunk 3]
```diff
@@ -98,6 +109,53 @@ void testFindByIpInterfaceForAGivenLocationAndIpAddress() throws UnknownHostExce
...
+        ipInterface.setTenantId(node.getTenantId());
+        ipInterface.setIpAddress(InetAddress.getByName("127.0.0.1"));
+        ipInterfaceRepository.save(ipInterface);
+        var result = ipInterfaceRepository.findById(ipInterface.getId());
+        assertThat(result).isPresent().get()
+            .extracting(i -> i.getIpAddress().getHostAddress(), i -> i.getSnmpInterface().getIfName())
+            .containsExactly("127.0.0.1", "test-snmp");
+        //test after snmp being deleted
+        snmpInterfaceRepository.deleteById(snmpInterface.getId());
+        var result2 = ipInterfaceRepository.findById(ipInterface.getId());
```
Comment (james8hut) [inline inventory/src/test/java/org/opennms/horizon/inventory/repository/IpInterfaceRepositoryTest.java:134]: Here, without the new code, findById won't get anything back. (Essentially, this is the case where a different tenant calls findById with somebody elses id.)

File: inventory/src/test/java/org/opennms/horizon/inventory/repository/IpInterfaceRepositoryTest.java [hunk 4]
```diff
@@ -98,6 +109,53 @@ void testFindByIpInterfaceForAGivenLocationAndIpAddress() throws UnknownHostExce
...
+        assertThat(result2).isPresent();
+    }
+
+    @Test
+    void testIpInterfaceWithSNMPIfDeleteIPIf() throws UnknownHostException {
+        int ifIndex = 10;
```
Comment (james8hut) [inline inventory/src/test/java/org/opennms/horizon/inventory/repository/IpInterfaceRepositoryTest.java:140]: similar

**Review by james8hut — CHANGES_REQUESTED** [review]
Please rebase on develop

File: inventory/src/test/java/org/opennms/horizon/inventory/repository/IpInterfaceRepositoryTest.java (hunk 1 above)
Comment (yangli2014) [reply inventory/src/test/java/org/opennms/horizon/inventory/repository/IpInterfaceRepositoryTest.java]: updated

File: shared-lib/protobuf/src/main/proto/nodescan.proto [hunk 5]
```diff
@@ -72,13 +73,12 @@ message SnmpInterfaceResult {
   int32 ifAdminStatus = 7;
   int32 ifOperatorStatus = 8;
   string ifAlias = 9;
-  string ipAddress = 10;
 }
 
-message IpTableScanResult {
+/*message IpTableScanResult {
```
Comment (james8hut) [inline shared-lib/protobuf/src/main/proto/nodescan.proto:78]: Query: Should this be removed rather than commented out?

**Review by james8hut — APPROVED** [review]

**Pull request merged**